src/Agent.Worker/ResourceMetricsManager.cs

```diff
@@ -286,65 +286,26 @@ private async Task GetMemoryInfoAsync()
 
             if (PlatformUtil.RunningOnLinux)
             {
-                // Some compact Linux distributions like UBI may not have "free" utility installed, or it may have a custom output
-                // We don't want to break currently existing pipelines with ADO warnings
-                // so related errors thrown here will be sent to the trace or debug logs by caller methods
-
-                using var processInvoker = HostContext.CreateService<IProcessInvoker>();
-
                 using var timeoutTokenSource = new CancellationTokenSource();
                 timeoutTokenSource.CancelAfter(TimeSpan.FromMilliseconds(METRICS_UPDATE_INTERVAL));
 
                 using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(
                     _context.CancellationToken,
                     timeoutTokenSource.Token);
 
-                processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                await Task.Run(() =>
                 {
-                    if (!message.Data.StartsWith("Mem:"))
-                    {
-                        return;
-                    }
-
-                    var processInvokerOutputString = message.Data;
-                    var memoryInfoString = processInvokerOutputString.Split(" ", StringSplitOptions.RemoveEmptyEntries);
-
-                    if (memoryInfoString.Length != 7)
-                    {
-                        throw new Exception("\"free\" utility has non-default output");
-                    }
+                    string memoryInfo = File.ReadAllText("/proc/meminfo");
+                    int totalMemory = int.Parse(memoryInfo.Split('\n')[0].Split(" ", StringSplitOptions.RemoveEmptyEntries)[1]);
+                    int freeMemory = int.Parse(memoryInfo.Split('\n')[1].Split(" ", StringSplitOptions.RemoveEmptyEntries)[1]);
```

Comment:
you do `memoryInfo.Split('\n')` twice here. Consider reusing it in variable to reduce mem allocations

Reply:
Thanks! Fixed.